### PR TITLE
[#1228] Only enable Next button when passphrase is valid

### LINKED ISSUE #1228: Add validation for Seed Phrase in the Restore Access flow
### User Story

As a user I want an ability to restore my account using my previously generated secure phrase

### Description

*Type*: Bug

*Summary*:  Secure Pharse field is only checked for NOT NULL value. No other validation present

#### Expected behavior
[comment]: Add the following validation: 
- exact count of words entered in Secure Phrase field should be 12
- no dot, comma, numbers, special characters entered

#### Actual behavior
[comment]: the Secure Pharse field is only checked for NOT NULL value. Once any information is entered user is able to proceed

### Reproduction

https://drive.google.com/open?id=0B7h1U1joPifRQ242OGZ1cXZyWHM

Open Status
Set up a password
Remember the secure phrase
Tap Switch Account
Tap Restore Access
Enter anything in the Secure Phrase field and enter password
Tap Restore Access

### Additional Information
Operating System: Android
Status version: 0.9.7+ (http://artifacts.status.im:8081/artifactory/nightlies-local/im.status.ethereum-e16b93.apk)
[2017_05_20_23_50_32 - logcat.txt](https://github.com/status-im/status-react/files/1016506/2017_05_20_23_50_32.-.logcat.txt)

### PR DESCRIPTION
**Pull request by jeluard**
fixes #1228

### Summary:

Only enable recover `next` action if the passphrase is sound.

### Steps to test:
- Open Status
- Recover an account
- Validate incorrect passphrase are not accepted

status: ready

### CODE REVIEW

**Review by goranjovic — APPROVED** [review]

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/FDcnPw for branch PR-4497

<blockquote><div><strong><a href="https://i.diawi.com/FDcnPw">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/QJdGWN for branch PR-4497

<blockquote><div><strong><a href="https://i.diawi.com/QJdGWN">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Review by flexsurfer — APPROVED** [review]

File: test/cljs/status_im/test/utils/ethereum/core.cljs [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
   (testing "ERC20 balance-of params"
     (let [contract "0x29b5f6efad2ad701952dfde9f29c960b5d6199c5"
           address "0xa7cfd581060ec66414790691681732db249502bd"]
-      (is (= (core/call-params contract "balanceOf(address)" address)
+      (is (= (ethereum/call-params contract "balanceOf(address)" address)
              {:to "0x29b5f6efad2ad701952dfde9f29c960b5d6199c5"
               :data "0x70a08231000000000000000000000000a7cfd581060ec66414790691681732db249502bd"})))))
+
+(deftest valid-words?
+  (is (not (true? (ethereum/valid-words? ["rate" "rate"]))))
```
Comment (rasom) [inline test/cljs/status_im/test/utils/ethereum/core.cljs:14]: `(not (true? ..` == `(false? ..`

Comment (jeluard) [reply test/cljs/status_im/test/utils/ethereum/core.cljs:14]: Both are different in general. `nil` could be returned for instance.
See concepts of truthy and falsey: http://blog.jayfields.com/2011/02/clojure-truthy-and-falsey.html

**Comment by Serhy** [comment]
@jeluard Buids  https://i.diawi.com/8iU2NG and https://i.diawi.com/238WqU has been tested on Android/IOS. 

1) Major regression with this changes at the moment is that previous accounts (created in 0.9.19) recovers different accounts using the same passphrase + password than in Develop build.
The `spy fly certain occur scale normal vague evolve decide rail advice asset` / `aaaaaa` should receover `Wiry Sunny Gnatcatter` while with above builds it's a 'Expensice Shady Hoki' (and different public key / wallet key)

2) Can not recover more than one account having the same password for both accounts in the same app. If I try to recover the second account it logs me into the first account. To reproduce:
a) Open Status and recover `spy fly certain occur scale normal vague evolve decide rail advice asset` / `aaaaaa` account
b) Log out and recover the secont account `alarm vanish box canyon garlic six clean cross impulse disagree planet proud` / `aaaaaa`
c) Check whether you have two accounts on Sign in page now (or check if profile of the second recovered account is different than first)

3) App hungs up on the 'Sign in to another account' with infinite 'Signing you in' figwheel when recovering second account with the different password than first account. To reproduce:
a) Open Status and recover `spy fly certain occur scale normal vague evolve decide rail advice asset` / `aaaaaa` account
b) Log out and recover the secont account `alarm vanish box canyon garlic six clean cross impulse disagree planet proud` / `bbbbbb`
**Actual result:** (From logcat `Handling re-frame event:  :account-recovered {"address":"0xDA95860c409b42e9e527F92E9B0cd24D60227c6b","pubkey":"","error":"could not decrypt key with given passphrase"}
` 
TF session for the last issue: https://app.testfairy.com/projects/4803622-status/builds/8421300/sessions/380/?accessToken=MKI-NRqEN6Yhe4WcCny2l2aNHEs

<blockquote><div><strong><a href="https://i.diawi.com/8iU2NG">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/238WqU">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803622-status/builds/8421300/sessions/380/?accessToken=MKI-NRqEN6Yhe4WcCny2l2aNHEs">TestFairy: Status 0.9.19 (6027) @380</a></strong></div></blockquote>

**Comment by jeluard** [comment]
@Serhy Good catch! Just pushed a fix.

**Comment by Serhy** [comment]
Looks good now! No regressions. I'm able to recover fine my previously used accounts.
Validation of the passphrase input is present as well: no dot/comma/number are allowed ('Sign In' button inactive in that case. Whitespaces got trimmed. 
Thanks, @jeluard, - ready for merge!

**Pull request merged**